ipa-core/src/protocol/basics/reveal.rs

```diff
@@ -206,6 +210,20 @@ impl<'a, F: ExtendableField> Reveal<UpgradedMaliciousContext<'a, F>, 1> for Mali
     }
 }
 
+// Workaround for https://github.com/rust-lang/rust/issues/100013. Calling this wrapper function
+// instead of `Reveal::reveal` seems to hide the `impl Future` GAT.
+pub fn reveal<'fut, C, S>(
```

Comment:
just curious if using `pub async fn reveal(...) -> Result<S::Output,..>` works too

Reply:
It does not... my hypothesis is it's related to this issue noted in my comment on #909:

> The compiler is smart enough to see that if you have &local in the async move future, it can make that work by moving local into the future. However, this often results in the HRTB errors when the compiler needs to prove a statement of the form "for any lifetime 'local describing &local in the anonymous future, some other anonymous lifetime like 'ctx must outlive 'local.", which looks like `for <'local> 'ctx: 'local`, which the compiler can only reduce to `'ctx: 'static`.